android/translate-showcase/app/src/main/java/com/google/mlkit/showcase/translate/main/MainFragment.kt

```diff
@@ -304,36 +296,21 @@ class MainFragment : Fragment() {
         holder.unlockCanvasAndPost(canvas)
     }
 
-    /**
-     * We need a display listener for orientation changes that do not trigger a configuration
-     * change, for example if we choose to override config change in manifest or for 180-degree
-     * orientation changes.
-     */
-    private val displayListener = object : DisplayManager.DisplayListener {
-        override fun onDisplayAdded(displayId: Int) = Unit
-        override fun onDisplayRemoved(displayId: Int) = Unit
-        override fun onDisplayChanged(displayId: Int) = view?.let { view ->
-            if (displayId == this@MainFragment.displayId) {
-                Log.d(TAG, "Rotation changed: ${view.display.rotation}")
-                imageAnalyzer?.targetRotation = view.display.rotation
-            }
-        } ?: Unit
-    }
-
     /**
      *  [androidx.camera.core.ImageAnalysisConfig] requires enum value of
      *  [androidx.camera.core.AspectRatio]. Currently it has values of 4:3 & 16:9.
      *
-     *  Detecting the most suitable ratio for dimensions provided in @params by counting absolute
+     *  Detecting the most suitable ratio for dimensions provided in @params by comparing absolute
      *  of preview ratio to one of the provided values.
      *
      *  @param width - preview width
      *  @param height - preview height
      *  @return suitable aspect ratio
      */
     private fun aspectRatio(width: Int, height: Int): Int {
-        val previewRatio = max(width, height).toDouble() / min(width, height)
-        if (abs(previewRatio - RATIO_4_3_VALUE) <= abs(previewRatio - RATIO_16_9_VALUE)) {
+        val previewRatio = log(max(width, height).toDouble() / min(width, height), 10.0)
```

Comment:
Nit: It doesn't really matter which logarithm base you use, so I would use the default natural logarithm, because it's shorter.